jobsdb/jobsdb.go

```diff
@@ -2373,8 +2378,21 @@ func (jd *HandleT) doStoreJobsInTx(ctx context.Context, tx *sql.Tx, ds dataSetT,
 		return err
 	}
 	err := store()
+
 	var e *pq.Error
 	if err != nil && errors.As(err, &e) {
+
+		if e.Code == lockTableInsertExceptionCode {
+			var updatedList []dataSetT
+			jd.dsListLock.WithLock(func(l lock.DSListLockToken) {
```

Comment:
great catch, it completely got out of my mind that we are calling it inside `WithStoreSafeTx` and this can create a deadlock situation. 
BTW, what will be an ideal (systematic) approach to avoid mistakes like this in future?